ci/entrypoint.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+set -euo pipefail
+
+# Add local zenbuilder user, either use LOCAL_USER_ID:LOCAL_GRP_ID
+# if set via environment or fallback to 9001:9001
+USER_ID="${LOCAL_USER_ID:-9001}"
+GRP_ID="${LOCAL_GRP_ID:-9001}"
+DOCKER_BUILD_DIR="${DOCKER_BUILD_DIR:-/build}"
+DOCKER_CARGO_HOME="${DOCKER_CARGO_HOME:-/tmp/.cargo}"
+CARGO_BINARIES_INSTALL="${CARGO_BINARIES_INSTALL:-}"
+NODEJS_VERSION_INSTALL="${NODEJS_VERSION_INSTALL:-}"
+TARGET_DIR="${DOCKER_BUILD_DIR}/target"
+
+fn_die() {
+  echo -e "\n\033[1;31m${1}\033[0m\n" >&2
+  exit "${2:-1}"
+}
+
+# Using generic cargo home dir location. 'CARGO_HOME' is a special env var that defines home dir for CARGO
+if [ -n "${DOCKER_CARGO_HOME}" ]; then
+  export CARGO_HOME="${DOCKER_CARGO_HOME}"
+fi
+
+# Run user mgmt
+if [ "$USER_ID" != "0" ]; then
+  export USERNAME=zenbuilder
+  getent group "$GRP_ID" &> /dev/null || groupadd -g "$GRP_ID" "$USERNAME"
+  id -u "$USERNAME" &> /dev/null || useradd --shell /bin/bash -u "$USER_ID" -g "$GRP_ID" -o -c "" -m "$USERNAME"
+  CURRENT_UID="$(id -u "$USERNAME")"
+  CURRENT_GID="$(id -g "$USERNAME")"
+  export HOME=/home/"$USERNAME"
+  if [ "$USER_ID" != "$CURRENT_UID" ] || [ "$GRP_ID" != "$CURRENT_GID" ]; then
+    echo -e "\nWARNING: User with differing UID ${CURRENT_UID}/GID ${CURRENT_GID} already exists, most likely this container was started before with a different UID/GID. Re-create it to change UID/GID.\n"
+  fi
+else
+  export USERNAME=root
+  export HOME=/root
+  CURRENT_UID="$USER_ID"
+  CURRENT_GID="$GRP_ID"
+  echo -e "\nWARNING: Starting container processes as root. This has some security implications and goes against docker best practice.\n"
+fi
+
+# Installing extra dependencies if required
+if [ -n "${CARGO_BINARIES_INSTALL}" ]; then
+  echo -e "\nInstalling extra cargo binaries: ${CARGO_BINARIES_INSTALL}\n"
+  for binary in $(tr "," " " <<< "${CARGO_BINARIES_INSTALL}"); do
+    cargo install --force "${binary}" || fn_die "ERROR: Failed to install cargo binary: ${binary}"
+  done
+fi
+
+# Node.js install if required
+if [ -n "${NODEJS_VERSION_INSTALL}" ]; then
+  echo -e "\n=== Installing Node.js version: ${NODEJS_VERSION_INSTALL} ===\n"
+  mkdir -p /etc/apt/keyrings
+  curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+  echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_${NODEJS_VERSION_INSTALL}.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list
+  export DEBIAN_FRONTEND=noninteractive
+  apt update -qq
+  apt --no-install-recommends install -y nodejs
+  npm install -g yarn
+  echo -e "Nodejs environment was successfully deployed. Node.js version: $(node -v) | npm version: $(npm -v) | yarn version: $(yarn -v)\n"
+fi
+
+# System info
+rustup show
+if command -v java; then
+  echo -e "\n$(java --version)"
+fi
```

Comment:
Very minor comment, I think this part can be revomed.

Reply:
This is not needed correct.